app/src/main/java/com/azuresamples/msalandroidapp/StaticTokenAuthProvider.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.azuresamples.msalandroidapp;
+
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import com.microsoft.graph.authentication.BaseAuthenticationProvider;
+
+import java.net.URL;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Implementation sample for a simplistic authentication provider
+ * <p>
+ * The Microsoft Graph SDK client accepts an authentication provider
+ * that is responsible for providing access tokens to authenticate API
+ * calls. Because this sample focuses on acquiring tokens manually, this
+ * provider simply returns the provided token.
+ * <p>
+ * Real-world providers would take advantage of MSAL's caching abilities
+ * to always get a valid token, refresh the token when expired, etc.
+ */
+public class StaticTokenAuthProvider extends BaseAuthenticationProvider {
+
+    private final String mAccessToken;
+
+    public StaticTokenAuthProvider(String accessToken) {
+        mAccessToken = accessToken;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    @Nonnull
```

Comment:
```suggestion
    @Nullable
```

Since this method _can_ return `null`

Reply:
Hmm - maybe it would be better to return `CompletableFuture.completedFuture("")`. @baywet what would the SDK handle more gracefully?

Reply:
yep, return a CompletableFuture.completedFuture(null) https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/dev/src/main/java/com/microsoft/graph/httpcore/AuthenticationHandler.java#L54

Reply:
Done